### PR TITLE
Update Optimization Performance docs

### LINKED ISSUE #4611: [Docs] provide clarity on the cache that flyte propeller uses
### Description

reference slack thread: https://discuss.flyte.org/t/16095263/please-could-someone-explain-which-cache-the-docs-are-referr#a02e379e-89dd-4fd9-ab49-df6b29bbcb65

doc page to update: https://docs.flyte.org/en/latest/deployment/configuration/performance.html#optimize-flytepropeller-configuration

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by davidmirror-ops**
## Tracking issue

Closes #4611 

## Why are the changes needed?
This PR looks to bring clarity and guide users to improve Propeller performance from a better understanding of how it works.

## Check all the applicable boxes 

- [X] I updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] All commits are signed-off.

### CODE REVIEW

**Review by neverett — COMMENTED** [review]
Left some questions and feedback, let me know if I can help clarify anything!

File: docs/deployment/configuration/performance.rst [hunk 1]
```diff
@@ -6,148 +6,211 @@ Optimizing Performance
...
 .. code-block:: yaml
 
     propeller:
-        kube-client-config:
-            qps: 100 # Refers to max rate of requests to KubeAPI server
-            burst: 50 # refers to max burst rate to Kube API server
-            timeout: 30s # Refers to timeout when talking with kubeapi server
+      kube-client-config:
+        qps: 100 # Refers to max rate of requests (queries per second) to kube-apiserver
+        burst: 25 # refers to max burst rate.
```
Comment (hamersaw) [inline docs/deployment/configuration/performance.rst:81]: burst should be >= qps. we did a little dive here into the code and realized we have been misconfiguring this for awhile.

File: docs/deployment/configuration/performance.rst [hunk 2]
```diff
@@ -167,37 +230,32 @@ This can done in multiple ways
...
-
 Sharded scale-out
 -------------------
-FlytePropeller Manager is a new component introduced as part of `this RFC <https://github.com/flyteorg/flyte/blob/master/rfc/system/1483-flytepropeller-horizontal-scaling.md>`_ to facilitate horizontal scaling of FlytePropeller through sharding. Effectively, the Manager is responsible for maintaining liveness and proper configuration over a collection of FlytePropeller instances. This scheme uses k8s label selectors to deterministically assign FlyteWorkflow CRD responsibilities to FlytePropeller instances, effectively distributing processing load over the shards.
+FlytePropeller Manager is a new component introduced to facilitate horizontal scaling of FlytePropeller through sharding. Effectively, the Manager is responsible for maintaining liveness and proper configuration over a collection of FlytePropeller instances. This scheme uses K8s label selectors to deterministically assign FlyteWorkflow CRD responsibilities to FlytePropeller instances, effectively distributing processing load over the shards.
```
[the remark this answers is not in the document]
Comment (hamersaw) [reply docs/deployment/configuration/performance.rst:252]: +1

**Review by hamersaw — COMMENTED** [review]
looks accurate to me! thanks for cleaning this up - it feels much more useful.

File: docs/deployment/configuration/performance.rst [hunk 3]
```diff
@@ -6,148 +6,211 @@ Optimizing Performance
...
-..  youtube:: FJ-rG9lZDhY 
+There are some base design attributes and assumptions that FlytePropeller applies:
 
-Scaling up FlytePropeller
-==========================
-`FlytePropeller <https://pkg.go.dev/github.com/flyteorg/FlytePropeller>`_ is the core engine of Flyte that executes the workflows for Flyte.
-It is designed as a `Kubernetes Controller <https://kubernetes.io/docs/concepts/architecture/controller/>`_, where the desired state is specified as a FlyteWorkflow `Custom Resource <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>`_.
+a. Every workflow execution is independent and can be performed by a completeley distinct process.
+b. When a workflow definition is compiled, the resulting DAG structure is traversed by the controller and the goal is to gracefully transition each task to Success.
+c. Node executions are performed by various FlytePlugins; a diverse collection of operations spanning Kubernetes and other remote services. FlytePropeller is only responsible for effectively monitoring and managing these executions.
```
[the remark this answers is not in the document]
Comment (davidmirror-ops) [reply docs/deployment/configuration/performance.rst:18]: Yeah,this was was inherited from the old doc but not very clear. I'll try to improve it.

File: docs/deployment/configuration/performance.rst (hunk 1 above)
Comment (davidmirror-ops) [reply docs/deployment/configuration/performance.rst:81]: Uh good finding. I tried to rephrase this section, to the best of my knowledge. I think I'll need to also update the configuration reference page in a separate PR.

File: docs/deployment/configuration/performance.rst [hunk 4]
```diff
@@ -6,148 +6,211 @@ Optimizing Performance
...
+     - Number of processes that can work concurrently. Also implies number of workflows that can be executed in parallel. Since FlytePropeller uses ``goroutines``, it can accommodate significantly more processes than the number of physical cores.
+     - ``flyte:propeller:all:free_workers_count``
+     -  A low number may result in higher overall latency for each workflow evaluation round. Larger the number, implies more workflows can be evaluated in parallel. But it should depend on number of CPU cores assigned to FlytePropeller and evaluated against the cost of context switching. A number around 500 - 800 with 4-8 CPU cores usually works fine.
```
[the remark this answers is not in the document]
Comment (davidmirror-ops) [reply docs/deployment/configuration/performance.rst:61]: thank you!

**Review by neverett — APPROVED** [review]

**Pull request merged**